File: app/src/main/java/com/BeeDocs/BeeDocsApplication.java
package com.BeeDocs;

import android.app.Application;
import android.content.Context;
import android.text.TextUtils;

import com.BeeDocs.model.LruBitmapCache;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.ImageLoader;
import com.android.volley.toolbox.Volley;

public class BeeDocsApplication extends Application {

	public static final String TAG = BeeDocsApplication.class
			.getSimpleName();

	private RequestQueue mRequestQueue;
	private ImageLoader mImageLoader;

	private static BeeDocsApplication mInstance;

	@Override
	public void onCreate() {
        super.onCreate();
        mInstance = this;
    }
   
	public static synchronized BeeDocsApplication getInstance() {
		return mInstance;
	}
	
	public static Context SuperContext() {
		return mInstance.getApplicationContext();
	}
	
	public RequestQueue getRequestQueue() {
		if (mRequestQueue == null) {
			mRequestQueue = Volley.newRequestQueue(getApplicationContext());
		}

		return mRequestQueue;
	}

	public ImageLoader getImageLoader() {
		getRequestQueue();
		if (mImageLoader == null) {
			mImageLoader = new ImageLoader(this.mRequestQueue,
					new LruBitmapCache());
		}
		return this.mImageLoader;
	}

	public <T> void addToRequestQueue(Request<T> req, String tag) {
		// set the default tag if tag is empty
		req.setTag(TextUtils.isEmpty(tag) ? TAG : tag);
		getRequestQueue().add(req);
	}

	public <T> void addToRequestQueue(Request<T> req) {
		req.setTag(TAG);
		getRequestQueue().add(req);
	}

	public void cancelPendingRequests(Object tag) {
		if (mRequestQueue != null) {
			mRequestQueue.cancelAll(tag);
		}
	}
}

File: app/src/main/java/com/beedocs/BeeDocsApplication.kt
package com.beedocs

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import com.beedocs.network.VolleyConnection

class BeeDocsApplication : Application() {
	
	val TAG = this.javaClass.simpleName
	
	private lateinit var volleyConnection: VolleyConnection
	
	override fun onCreate() {
		super.onCreate()
		instance = this
		volleyConnection = VolleyConnection(getContext())
	}
	
	fun getVolleyConnection() : VolleyConnection {
		return volleyConnection
	}
	
	companion object {
		@get:Synchronized
		var instance: BeeDocsApplication? = null
			private set
		
		@JvmStatic
		fun getContext(): Context {
			return instance!!.applicationContext
		}
	}
}